### DISCUSSION TITLE
Front/Back image on device overview + Legenda from roles

### DISCUSSION THREAD

**Discussion by deku-m** [netbox-community/netbox#11643] [Ideas]
Today i was working on devices and looked into front/back image, if it was possible to include the device front/back image on the device overview itself.? Now it is only attached and used in the rack coming straight from the device-types. I couldnt get it on the device overview page as it isnt provided:

**Device-type:**
![afbeelding](https://user-images.githubusercontent.com/37069737/216319370-49f4a29d-c3a4-4c3d-8e1d-aa05942125c8.png)
**Device (overview)**
![afbeelding](https://user-images.githubusercontent.com/37069737/216319396-79b34f68-ad01-487b-90fd-72ffc6aac733.png)
I think it might be usefull to have this also on the device overview itself coming straight from the device-type perhaps.

Regarding our device image design, we created the following:
As we got a simple idea to just use a design (no copyright and simple with ports showing) like this below. 
We created it in a simple excel document as it gives an example of the port allocation by color which you could bind to roles or something.
![afbeelding](https://user-images.githubusercontent.com/37069737/216319430-8efaf209-29e7-43bc-9330-faa652191e1c.png)
![afbeelding](https://user-images.githubusercontent.com/37069737/216321289-b880cbfb-d3c8-414d-8788-05f48fa79aab.png)

**Legend:**
I wonder if its possible to include some sort of a colored legenda on the device overview page like in the excel design example above.
Tried it with a custom field as I wanted to reuse the device roles for my example, as they already contain the colors.
But havent succeeded.
![afbeelding](https://user-images.githubusercontent.com/37069737/216319684-b1c7c176-5f01-48cf-a546-0b55ede3ab45.png)

**Custom-Field:**
Is it also possible to change custom field title in the future if that is even possible.?

**Comment by deku-m** [comment]
@jeremystretch I wonder what you think or others ofc?
Is it something I need to create feature requests for?

**Reply by candlerb** [reply]
You didn't say what version of Netbox you're using, but in v3.4.4, if the device is racked, the device view at `/dcim/devices/NNN/` now shows an image of the rack at the bottom right hand side, with the current device highlighted and all the others greyed out, like this (for device "gw2"):

![image](https://user-images.githubusercontent.com/44789/217856093-73cc306f-41a2-45a6-b677-a1c55fbe14dc.png)

I don't know exactly when this feature was added, but it seems to cover your first requirement of wanting to see the front and rear images from the device type, when viewing the device.

The second part of your feature request (being able to colour the ports depending on their usage) is much harder, as it would require mapping every port to a rectangle on the device type.

> Is it also possible to change custom field title in the future if that is even possible.?

Yes, you can change the "Label" for a custom field, which is how the field is named on the web display (whereas the "Name" is the key used in the JSON custom_field_data, and you should not change this)

**Reply by deku-m** [reply]
I am running 3.4.4 in docker. But its more that i would like to see the full picture like what is given in the device type view.
You get two entries with front and back image (the bigger images as in the example) 
This one:
![](https://user-images.githubusercontent.com/37069737/216319370-49f4a29d-c3a4-4c3d-8e1d-aa05942125c8.png)

Dont want to really link with the device ports in netbox itself. We use the attached images as preview and can edit the port colours in excel for  what we created as images for front/back. If a Legenda could be added, we could match the documented colours with what is written in the legenda. Instead of creating a real mapped link in netbox.

I meant the Title named Custom Fields. As you create your own field under that so for example if i create a custom field named Legenda you get (like in the below example):
Custom Fields:  <<< This is what i hoped to change.
Legenda:

Example:
![](https://user-images.githubusercontent.com/37069737/216319684-b1c7c176-5f01-48cf-a546-0b55ede3ab45.png)
